lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -191,6 +192,113 @@ class BlockedToMFMA : public mlir::RewritePattern {
   }
 };
 
+class ChainDotAMD : public mlir::RewritePattern {
+
+public:
+  ChainDotAMD(mlir::MLIRContext *context)
+      : mlir::RewritePattern(triton::DotOp::getOperationName(), 2, context) {}
+
+  bool isChainDot(triton::DotOp &dotOp) const {
+    auto filter = [&dotOp](Operation *op) {
+      return op->getParentRegion() == dotOp->getParentRegion();
+    };
+
+    SetVector<Operation *> forwardSlices;
+    forwardSlices.clear();
+    mlir::getForwardSlice(dotOp.getResult(), &forwardSlices, filter);
+
+    bool chainDot = false;
+    for (Operation *op : forwardSlices) {
+      if (isa<triton::DotOp>(op) && (op != dotOp))
+        chainDot = true;
+    }
+    return chainDot;
+  }
+
+  bool hasView(triton::DotOp &dotOp) const {
+    // Check if we already optimized chain dot.
+    auto filter = [&dotOp](Operation *op) {
+      return op->getParentRegion() == dotOp->getParentRegion();
+    };
+    SetVector<Operation *> bwdSlices;
+    bwdSlices.clear();
+    mlir::getBackwardSlice(dotOp.getResult(), &bwdSlices, filter);
+
+    bool hasView = false;
+    for (Operation *op : bwdSlices) {
+      if (isa<mlir::triton::ViewOp>(op))
+        hasView = true;
+    }
+    return hasView;
+  }
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::Operation *op,
+                  mlir::PatternRewriter &rewriter) const override {
+    auto dotOp = cast<triton::DotOp>(op);
+
+    auto oldRetType = dotOp.getResult().getType().cast<RankedTensorType>();
+    auto operandType = dotOp.getA().getType().cast<RankedTensorType>();
+    if (!isChainDot(dotOp) || hasView(dotOp) ||
```

Comment:
Can you elaborate on this `hasView()` function? Why a ViewOp stops us from "transpose" the mfma layout?